File: src/main/java/cn/wisesign/xamng/MainSetup.java
package cn.wisesign.xamng;

import java.util.Date;

import org.nutz.dao.Dao;
import org.nutz.dao.util.Daos;
import org.nutz.ioc.Ioc;
import org.nutz.mvc.NutConfig;
import org.nutz.mvc.Setup;

import cn.wisesign.xamng.bean.Case;

public class MainSetup implements Setup {

    public void init(NutConfig conf) {
        Ioc ioc = conf.getIoc();
        Dao dao = ioc.get(Dao.class);
        Daos.createTablesInPackage(dao, "cn.wisesign.xamng", false);
        
        if (dao.count(Case.class) == 0) {
        	Case mycase = new Case();
        	mycase.setName("admin");
        	mycase.setBelongProject("123456");
        	mycase.setScript("123123");
        	mycase.setCreatedate(new Date());
            dao.insert(mycase);
        }
    }

    public void destroy(NutConfig conf) {
    }

}

File: src/main/java/cn/wisesign/xamng/MainSetup.kt
package cn.wisesign.xamng


import org.nutz.dao.Dao
import org.nutz.dao.util.Daos
import org.nutz.mvc.NutConfig
import org.nutz.mvc.Setup
import org.openqa.grid.internal.utils.configuration.GridHubConfiguration
import org.openqa.grid.web.Hub


class MainSetup():Setup {

    companion object{
        var selenium_server:Hub = Hub(GridHubConfiguration())
        fun booSeleniumServer(){
            var selenium_server_config = GridHubConfiguration()
            selenium_server_config.port=7474
            selenium_server = Hub(selenium_server_config)
            selenium_server.start()
        }
    }

    override fun init(conf:NutConfig) {
        var dao:Dao = conf.ioc.get(Dao::class.java)
        Daos.createTablesInPackage(dao, "cn.wisesign.xamng", false)
        booSeleniumServer()
    }

    override fun destroy(conf:NutConfig) {
    }

}
